pandas/plotting/_matplotlib/hist.py

```diff
@@ -417,11 +419,21 @@ def hist_frame(
         if not isinstance(column, (list, np.ndarray, ABCIndexClass)):
             column = [column]
         data = data[column]
-    data = data._get_numeric_data()
+    # GH32590
+    columns_copy = data.columns
+    numeric_data = data._get_numeric_data()
+    datetime_data = data.select_dtypes(include="datetime64[ns]")
```

Comment:
should only `datetime` type be included? what about `timedelta`? could you pls check if `timedelta` should also be supported?

if so you could do something like:
```python
include_type = ['datetime', 'timedelta']
datetime_data = data.select_dtypes(include=include_type)
```
or you could even do
```python
include_type = ['datetime', 'timedelta', np.number]
data = data.select_dtypes(include=include_type)
```
then probably you could avoid using `data._get_numeric_data()` and `concat`.

Reply:
Agree with the first point. 
Would it be ok to use `np.number` instead of `_get_numeric_data()`? I'm not sure about this.
I'll have a check on both.

Reply:
@charlesdong1991 @jreback `timedelta` is not supported even in `pd.Series.hist` but, we can convert both `timedelta` and `datetime[ns]` into `int64` then keep `data=data._get_numeric_data()`. Any thought?

Reply:
yeah, `timedelta` is treated as numbers since it represents interval, probably `_get_numeric_data` does some other processingss to filter them out, then I think you could keep `_get_numeric_data` and `concat` in this PR!